R/format_data.R

```diff
@@ -1,121 +1,162 @@
-#' Format Data for Pharmacokinetic Analysis
+#' Create PK Concentration Dataset
 #'
-#' This function formats a data file for pharmacokinetic analysis by converting specific
-#' concentration values to numeric and adjusting time based on dose number.
+#' This function creates a pharmacokinetic concentration dataset from the provided ADNCA data.
 #'
-#' @param datafile A data frame containing the raw data to be formatted.
+#' @param ADNCA A data frame containing the ADNCA data.
+#' @param analyte A character string specifying the analyte of interest.
+
 #'
-#' @return A data frame with formatted concentration and time values.
+#' @return A data frame containing the filtered and processed concentration data.
 #'
 #' @details
 #' The function performs the following steps:
 #' \itemize{
-#'   \item{
-#'      Converts concentration values ('AVAL') that are 'BLQ', 'Negative',
-#'      'negative', or 'NEGATIVE' to 0.
-#'    }
-#'   \item{
-#'     Adjusts the time values ('TIME') based on the dose number ('DOSNO'). If 'DOSNO' is 1,
-#'     'TIME' is set to 'AFRLT'; otherwise, it is set to 'TIME'.
-#'   }
+#'   \item Filters the data for the specified analyte and event ID (EVID == 0).
+#'   \item Creates a 'groups' column by concatenating 'USUBJID' and 'DOSNO'.
+#'   \item Filters out negative time values.
+#'   \item Arranges the data by 'STUDYID', 'USUBJID', 'PCSPEC', 'DOSNO', and 'TIME'.
+#'   \item Adds an index column ('IX') within each group.
 #' }
 #'
 #' @examples
 #' \dontrun{
 #'   # Example usage:
-#'   formatted_data <- format_data(raw_data)
+#'   conc_data <- create_conc(ADNCA, "analyte_name", "profile_type")
 #' }
 #'
 #' @import dplyr
 #' @export
-format_data <- function(datafile) {
-  datafile <- datafile %>%
-    mutate(
-      AVAL = as.numeric(ifelse(AVAL %in% c("BLQ", "Negative", "negative", "NEGATIVE"), 0, AVAL)),
-      TIME = ifelse(DOSNO == 1, AFRLT, TIME)
-    )
 
-  return(datafile)
+#create pknca concentration dataset
+create_conc <- function(ADNCA,
+                        group_columns,
+                        time_column = "AFRLT") {
+  if (nrow(ADNCA) == 0) {
+    stop("Input dataframe is empty. Please provide a valid ADNCA dataframe.")
+  }
+
+  missing_columns <- setdiff(c(group_columns, time_column), colnames(ADNCA))
+  if (length(missing_columns) > 0) {
+    stop(paste("Missing required columns:", paste(missing_columns, collapse = ", ")))
+  }
+
+  data <- ADNCA %>%
```

Comment:
issue: variable assignment not needed, can just return the pipe result
```suggestion
  ADNCA %>%
```